### PR TITLE
Allocate Tenant without TenantKey Fails in the ibm-fhir-schematool on db2 #2349

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-install/src/main/docker/ibm-fhir-schematool/tests/2349/2349.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+echo "testuser:password" | chpasswd
+su - db2inst1
+db2 drop db fhirdb
+db2 CREATE DB FHIRDB using codeset UTF-8 territory us PAGESIZE 32768
+db2 grant connect on database TO USER testuser
+
+# 2 - use tool / exercise without TENANT_KEY
+docker-compose run tool --tool.behavior=onboard --db.host=192.168.86.29 --db.port=50000 --user=db2inst1 --password=change-password --db.database=fhirdb --sslConnection=false --db.type=db2 --schema.name.fhir=CLINIC --grant.to=testuser     --tenant.name=CLINICA
+
+# Check the TENANTS table // Should be 2700+
```
Comment (punktilious) [inline fhir-install/src/main/docker/ibm-fhir-schematool/tests/2349/2349.sh:28]: What should be 2700+?

Comment (prb112) [reply fhir-install/src/main/docker/ibm-fhir-schematool/tests/2349/2349.sh:28]: Whoops left over from the prior test

Comment (prb112) [inline fhir-install/src/main/docker/ibm-fhir-schematool/tests/2349/2349.sh:28]:
```suggestion
# Check the TENANTS table
```

File: fhir-install/src/main/docker/ibm-fhir-schematool/run.sh [hunk 2]
```diff
@@ -242,49 +242,44 @@ function allocate_tenant {
...
+        else
+            # not specified by the client, so we're randomly generating and inlining it
+            TENANT_KEY=$(openssl rand -base64 32)
+            echo "TENANT_KEY: ${TENANT_KEY}"
+            echo "${TENANT_KEY}" > ${TK_FILE}
+            echo "tenant.key=${TENANT_KEY}" >> ${TOOL_INPUT_FILE}
         fi
 
+        # Only in the case where we have a file we are loading we hit the issue
+        # where we need to override the default behavior or exiting on failure.
```
Comment (lmsurpre) [inline fhir-install/src/main/docker/ibm-fhir-schematool/run.sh:258]: not sure i follow this comment, but I think its ok.
```suggestion
        # Only in the case where we have a file we are loading we hit the issue
        # where we need to override the default behavior of exiting on failure.
```

File: fhir-install/src/main/docker/ibm-fhir-schematool/run.sh [hunk 3]
```diff
@@ -242,49 +242,44 @@ function allocate_tenant {
...
         if [ -z "${SCHEMA_FHIR}" ] || [ "null" = "${SCHEMA_FHIR}" ]
         then
             SCHEMA_FHIR="FHIRDATA"
         fi
-        _call_db2 "--schema-name ${SCHEMA_FHIR} --allocate-tenant ${TENANT_NAME} ${TK_FILE_STANZA} --pool-size 1"
 
-        # Always reset
-        set -o errexit
-        set -o pipefail
+        _call_db2 "--schema-name ${SCHEMA_FHIR} --allocate-tenant ${TENANT_NAME} --tenant-key-file ${TK_FILE} --pool-size 5"
```
Comment (lmsurpre) [inline fhir-install/src/main/docker/ibm-fhir-schematool/run.sh:268]: I assume we'd done enough testing with a pool-size of 5 to feel good about this?

Comment (prb112) [reply fhir-install/src/main/docker/ibm-fhir-schematool/run.sh:268]: Yeah - I'm pretty confident. at least I haven't hit issues with size 5 yet

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**